common/src/main/java/io/netty/util/internal/logging/InternalLoggerFactory.java

```diff
@@ -77,7 +80,22 @@ InternalLogger newInstance(String name) {
 
     @SuppressWarnings("UnusedCatchParameter")
     private static InternalLoggerFactory newDefaultFactory(String name) {
-        InternalLoggerFactory f;
+        InternalLoggerFactory f = null;
+
+        String loggerType = SystemPropertyUtil.get("io.netty.logger.type");
+        if (loggerType != null) {
+            try {
+                Class<?> clazz = Class.forName(loggerType);
+                Field field = clazz.getField("INSTANCE");
+                f = (InternalLoggerFactory) field.get(null);
+                f.newInstance(name).debug("Using {} as the default logging framework", loggerType);
+            } catch (Throwable t) {
```

Comment:
Can we please not catch throwable here? Yes, it's done elsewhere in this class but I think those should be removed as part of #6096.